js/ace.js

```diff
@@ -0,0 +1,408 @@
+                'fees': 'https://helpcenter.ace.io/hc/zh-tw/articles/360018609132-%E8%B2%BB%E7%8E%87%E8%AA%AA%E6%98%8E',
+            },
+            'requiredCredentials': {
+                'apiKey': true,
+                'secret': true,
+                'phone': true,
+            },
+            'api': {
+                'public': {
+                    'get': [
+                        'list/tradePrice',
+                        'list/marketPair',
+                        'list/orderBooks/{base}/{quote}',
+                    ],
+                },
+                'private': {
+                    'post': [
+                        'v1/coin/customerAccount',
+                        'v1/kline/getKlineMin',
+                        'v1/order/order',
+                        'v1/order/cancel',
+                        'v1/order/getOrderList',
+                        'v1/order/showOrderStatus',
+                        'v1/order/showOrderHistory',
+                    ],
+                },
+            },
+            'fees': {
+                'trading': {
+                    'percentage': true,
+                    'maker': this.parseNumber ('0.0005'),
+                    'taker': this.parseNumber ('0.001'),
+                },
+            },
+            'options': {
+                'networks': {
+                    'ERC20': 'ERC20',
+                    'ETH': 'ERC20',
+                    'TRX': 'TRX',
+                    'TRC20': 'TRX',
+                },
+            },
+            'precisionMode': TICK_SIZE,
+            'exceptions': {
+                'exact': {
+                },
+                'broad': {
+                },
+            },
+            'commonCurrencies': {
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        /**
+         * @method
+         * @name ace#fetchMarkets
+         * @description retrieves data on all markets for ace
+         * @param {object} params extra parameters specific to the exchange api endpoint
+         * @returns {[object]} an array of objects representing market data
+         */
+        let response = await this.publicGetListMarketPair ();
+        // temporary workaround to finx json string
+        if (response.indexOf ('\'') >= 0) {
+            // this line didn't work in php
+            response = response.replace (/'/g, '"');
+            response = this.parseJson (response);
+        }
+        //
+        //     ['ADA/TWD']
+        //
+        const result = [];
+        for (let i = 0; i < response.length; i++) {
+            const id = response[i];
+            const splitMarket = id.split ('/');
+            const baseId = this.safeString (splitMarket, 0);
+            const quoteId = this.safeString (splitMarket, 1);
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const symbol = base + '/' + quote;
+            const limits = {
```

Comment:
I usually just put this inside the argument to `result.push`, it just reduces the overall amount of code, and if you don't need to use it more than once, there's no point in assigning a variable, and writing outside of the `result.push` object actually makes it more difficult to read